src/hooks/index.ts

```diff
@@ -869,3 +893,23 @@ export const useResync = (wallet: YoroiWallet, options?: UseMutationOptions<void
     resync: mutation.mutate,
   }
 }
+
+export const useNfts = (wallet: YoroiWallet) => {
+  const utxos = useUtxos(wallet)
+  const {data, ...rest} = useQuery({
+    queryKey: [wallet.id, 'nfts', utxos.length],
+    queryFn: () => wallet.fetchNfts(),
+  })
+
+  return {...rest, nfts: data ?? []}
+}
+
+export const useNft = (wallet: YoroiWallet, {id}: {id?: string} = {}): YoroiNFT => {
+  const {nfts} = useNfts(wallet)
+  const nft = nfts.find((nft) => nft.id === id)
+
+  if (!nft) {
+    throw new Error(`Invalid id used "${id}" to get NFT`)
```

Comment:
if the NFT is no longer available we will throw?

Reply:
yeah, as discussed with @wolverineks it would be better to throw so we don have to handle null and undefined